secure/net/src/network_controller/outbound_handler.rs

```diff
@@ -29,84 +30,127 @@ impl OutboundHandler {
         inbound_handler: Arc<Mutex<InboundHandler>>,
     ) -> Self {
         Self {
-            service,
-            network_clients: Arc::new(Mutex::new(HashMap::new())),
+            _service: service,
+            remote_addresses: HashSet::new(),
             address: listen_addr,
-            handlers: Arc::new(Mutex::new(Vec::new())),
+            handlers: Vec::new(),
             inbound_handler,
         }
     }
 
     pub fn register_handler(
-        &self,
+        &mut self,
         message_type: String,
         remote_addr: SocketAddr,
         receiver: Receiver<Message>,
     ) {
-        // Create a remote client if it doesn't exist
-        self.network_clients
-            .lock()
-            .unwrap()
-            .entry(remote_addr)
-            .or_insert_with(|| NetworkClient::new(message_type.clone(), remote_addr, 5000));
-        let mut handlers = self.handlers.lock().unwrap();
-        handlers.push((receiver, remote_addr, MessageType::new(message_type)));
+        self.remote_addresses.insert(remote_addr);
+        self.handlers
+            .push((receiver, remote_addr, MessageType::new(message_type)));
     }
 
-    pub fn start(&mut self) {
-        let outbound_handlers = self.handlers.clone();
+    pub fn start(&mut self, rt: &Runtime) -> Option<Sender<Message>> {
+        if self.handlers.is_empty() {
+            return None;
+        }
+
+        // Register a signal handler to stop the outbound task
+        let (stop_signal_tx, stop_signal_rx) = unbounded();
+        self.handlers.push((
```

Comment:
Better you do it in the constructor and remove the handler.is_empty() check on line 53.

Reply:
That will require all the handlers registered in the constructor ? Will that not break the nice modularity that we have ? That is 1 network controller/outbound_task per node. And many modules are individually register their handlers